source/agora/consensus/PreImage.d

```diff
@@ -284,19 +286,20 @@ unittest
 
 public struct PreImageCycle
 {
+    @safe nothrow:
+
     /// Make sure we get initializedby disabling the default ctor
     @disable public this ();
 
-    /// Construct an instance with the provided cycle length
-    public this (in Scalar secret, uint cycle_length) @safe nothrow
+    /// Construct an instance with the provided cycle length and number of cycles
+    public this (in Scalar secret, uint cycle_length, const ulong cycles = NumberOfCycles,
```

Comment:
I think we don't need the `cycles` parameter because there are few cases where we need to change the length of cycles. Just use `NumberOfCycles` please.

Reply:
This is to decrease the time needed for the the test below which can test for multiple batches of `pre-image`s and make sure the implementation of using the `nonce` is correct.